### PR TITLE
Semantic Versioning

### LINKED ISSUE #242: Use Semantic Versioning
**Description**:
We've discussed a few times that the API should use semantic versioning, certainly by release time. This has caused a couple of minor deployment issues - the version is checked during deployment and because it's always the same, the API often isn't reinstalled by default.

I don't know much about this, but I did quickly scan a fairly useful article a few days ago on this subject - https://mestrak.com/blog/semantic-release-with-python-poetry-github-actions-20nn

**Acceptance criteria**:
- [ ] Start using semantic versioning on DataGateway API
- [ ] Document the process of how and when to increase the version of the API

### PR DESCRIPTION
**Pull request by MRichards99**
This PR will close #242 

## Description
This PR contains changes for the API's versioning. Once this PR is merged, the API will follow semantic versioning, using a couple of automated tools to manage this. This is explained with the additions in `README.md` but essentially involves using the Angular commit message style to represent what type of release/version bump should be made. I've also added an extra check box for the PR template just as a reminder when you create/review a PR.

Since DataGateway API will be going into production soon and it's essentially ready for this, I have added a `BREAKING CHANGE:` commit which will automatically bump the API to v1.0.0.

This branch also contains commits from #275 to get the CI to pass.

## Testing Instructions
To test, I have forked DataGateway API (https://github.com/MRichards99/datagateway-api), created a [PR equivalent to this](https://github.com/MRichards99/datagateway-api/pull/3) and merged it. You can see that a[ GitHub release of v1.0.0](https://github.com/MRichards99/datagateway-api/releases) has been made on my fork, as intended. Have a look at this and check everything looks all good, or fork DataGateway API yourself and follow the steps I did to see if it works for you.

- [ ] Review code
- [ ] Check GitHub Actions build
- [ ] If `icatdb Generator Script Consistency Test` CI job fails, is this because of a
      deliberate change made to the script to change generated data (which isn't
      actually a problem) or is there an underlying issue with the changes made?
- [ ] Review changes to test coverage

## Agile Board Tracking
Connect to #242

### CODE REVIEW

**Comment by MRichards99** [comment]
The safety part of the CI workflow will be fixed once https://github.com/ral-facilities/datagateway-api/pull/278 is merged.

File: README.md [hunk 1]
```diff
@@ -281,6 +282,59 @@ pre-commit install
...
+# Commit for a new feature
+feat: Add endpoints for search API #issue-number
+
+# Commit which introduces a breaking change for users
+BREAKING CHANGE: Change format of `config.json`, the previous version is no longer supported #issue-number
+
+# You can also use `BREAKING CHANGE:` in the additional information if the commit also adds a new feature, like so:
+feat: My new feature #issue-number
+
+BREAKING CHANGE: This feature means X functionality has been removed
```
Comment (VKTB) [inline README.md:322]: Angular's guide recommends not to capitalise the first letter of the subject so not sure if we want to follow that too, but suggesting anyway.

```suggestion
# Commit to edit a CI job
ci: edit linting job #issue-number

# Commit for a bug fix
fix: fix bug found with count endpoints #issue-number

# Commit for a new feature
feat: add endpoints for search API #issue-number

# Commit which introduces a breaking change for users
BREAKING CHANGE: change format of `config.json`, the previous version is no longer supported #issue-number

# You can also use `BREAKING CHANGE:` in the additional information if the commit also adds a new feature, like so:
feat: my new feature #issue-number

BREAKING CHANGE: this feature means X functionality has been removed
```

**Review by VKTB — COMMENTED** [review]
I like this. Looks good, just one slight thing I noticed which I outlined in the comment below.

File: README.md (hunk 1 above)
Comment (MRichards99) [reply README.md:322]: I hadn't noticed that, we should follow that to keep with Angular's format as much as possible.

**Review by VKTB — APPROVED** [review]

**Pull request merged**